### PR TITLE
fix: bugsnag event sent as an object for track call

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Resolves INT-752
For track call event sent as an object resulted in some alerts. This PR resolves those 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by mihir-4116 — DISMISSED** [review]

**Comment by sanpj2292** [comment]
@anantjain45823 
Let's write a common validator function for checking eventType. Ask everyone to only use it.

**Comment by anantjain45823** [comment]
We can have validation based upon message.type and basic request schema.
Does that sound good?

**Comment by sanpj2292** [comment]
>We can have validation based upon message.type and basic request schema.
Does that sound good?

Let us solve this problem via a common validator. The other thing of basic request schema validation, let us do it later.

**Comment by anantjain45823** [comment]
> > We can have validation based upon message.type and basic request schema.
> > Does that sound good?
> 
> Let us solve this problem via a common validator. The other thing of basic request schema validation, let us do it later.

Done

File: src/v0/util/index.js [hunk 1]
```diff
@@ -2025,6 +2025,11 @@ const getAuthErrCategoryFromStCode = (status) => {
   return '';
 };
 
+const validateEventType = event => {
+  if(!event || typeof event !== "string"){
+    throw new InstrumentationError("Event is a required field and should be a string");
```
Comment (sanpj2292) [inline src/v0/util/index.js:2030]: After throwing this, are we getting related statTags ? Can we once validate it ?

Comment (anantjain45823) [reply src/v0/util/index.js:2030]: Yes we are getting the stat tags as well. SS fro ref:

**Review by sanpj2292 — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**